### PR TITLE
Asgard deployment reporting

### PR DESCRIPTION
**Pull request by macdiesel**
@edx/pipeline-team PTAL. Deployments now report back about what ASGs were added and removed during a deploy.

### CODE REVIEW

File: scripts/asgard-deploy.py [hunk 1]
```diff
@@ -14,29 +14,39 @@
...
+        else:
+            click.echo('Would have triggered a deploy of {}'.format(ami_id))
+            deploy_info = {}
+
+        if out_file:
+            with open(out_file, 'w') as stream:
+                yaml.safe_dump(deploy_info, stream, default_flow_style=False, explicit_start=True)
         else:
-            click.echo("Would have triggered a deploy of {}".format(ami_id))
+            print yaml.dump(deploy_info, default_flow_style=False, explicit_start=True)
```
Comment (doctoryes) [inline scripts/asgard-deploy.py]: Why not `safe_dump()` here?

File: tubular/asgard.py [hunk 2]
```diff
@@ -508,14 +513,23 @@ def deploy(ami_id):
     LOG.info("ASG instances are healthy. Enabling Traffic.")
 
     elbs_to_monitor = []
+    current_asgs = defaultdict(list)  # Used to store the return value of what ASG's are currently deployed
     for cluster, asg in new_asgs.iteritems():
         try:
             enable_asg(asg)
             elbs_to_monitor.extend(elbs_for_asg(asg))
+            current_asgs['cluster'].append(asg)
```
Comment (doctoryes) [inline tubular/asgard.py]: Is this supposed to be:

```
current_asgs[cluster].append(asg)
```

?

File: tubular/asgard.py [hunk 3]
```diff
@@ -539,12 +553,16 @@ def deploy(ami_id):
...
+            if is_asg_enabled(asg):
+                disable_asg(asg)
+                disabled_asg[cluster].append(asg)
             try:
                 ec2.tag_asg_for_deletion(asg)
             except ASGDoesNotExistException as e:
                 LOG.info("Unable to tag ASG {} as it no longer exists, skipping".format(asg))
 
     LOG.info("Woot! Deploy Done!")
+    return {'current_asgs': current_asgs, 'disabled_asgs': disabled_asg}
```
Comment (doctoryes) [inline tubular/asgard.py:568]: Might not cause any problems - but a `defaultdict` isn't the same as a `dict` in some cases. Do you need to convert these to `dict`s here?

**Review by doctoryes — COMMENTED** [review]
1st pass complete.

File: tubular/asgard.py (hunk 3 above)
Comment (macdiesel) [reply tubular/asgard.py:568]: Really?

```
>>> a = defaultdict(list)
>>> a['item1'].extend([1,2,3,4])
>>> isinstance(a,dict)
True
```

File: scripts/asgard-deploy.py (hunk 1 above)
Comment (macdiesel) [reply scripts/asgard-deploy.py]: I was outputting to the screen vs a file, I changed it to be consistent.

File: tubular/asgard.py (hunk 2 above)
Comment (macdiesel) [reply tubular/asgard.py]: good catch.

File: tubular/asgard.py (hunk 3 above)
Comment (doctoryes) [reply tubular/asgard.py:568]: Now that you've called my bluff 😺 , I've messed around at the command line and can't remember in what context I experienced this. So I've been left with an unfounded superstition with no facts to back it up. So ignore this comment and proceed.

**Comment by doctoryes** [comment]
👍

**Pull request merged**